okhttp/src/main/java/com/squareup/okhttp/internal/DiskLruCache.java

```diff
@@ -190,7 +193,11 @@ public void run() {
   }
 
   // Visible for testing.
-  void initialize() throws IOException {
+  synchronized void initialize() throws IOException {
```

Comment:
Should this assume the caller already did synchronized? Maybe with a guard on like `Thread.holdsLock(this)` ?